service/src/main/java/bio/terra/axonserver/app/controller/CromwellWorkflowController.java

```diff
@@ -0,0 +1,145 @@
+package bio.terra.axonserver.app.controller;
+
+import bio.terra.axonserver.api.CromwellWorkflowApi;
+import bio.terra.axonserver.model.ApiWorkflowIdAndLabel;
+import bio.terra.axonserver.model.ApiWorkflowIdAndStatus;
+import bio.terra.axonserver.model.ApiWorkflowMetadataResponse;
+import bio.terra.axonserver.model.ApiWorkflowQueryResponse;
+import bio.terra.axonserver.service.cromwellworkflow.CromwellWorkflowService;
+import bio.terra.axonserver.service.wsm.WorkspaceManagerService;
+import bio.terra.common.exception.ApiException;
+import bio.terra.common.iam.BearerTokenFactory;
+import io.swagger.client.model.CromwellApiLabelsResponse;
+import io.swagger.client.model.CromwellApiWorkflowIdAndStatus;
+import io.swagger.client.model.CromwellApiWorkflowMetadataResponse;
+import io.swagger.client.model.CromwellApiWorkflowQueryResponse;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+import javax.annotation.Nullable;
+import javax.servlet.http.HttpServletRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+
+@Controller
+public class CromwellWorkflowController extends ControllerBase implements CromwellWorkflowApi {
+  private final CromwellWorkflowService cromwellWorkflowService;
+  private final WorkspaceManagerService wsmService;
+
+  @Autowired
+  public CromwellWorkflowController(
+      BearerTokenFactory bearerTokenFactory,
+      HttpServletRequest request,
+      CromwellWorkflowService cromwellWorkflowService,
+      WorkspaceManagerService wsmService) {
+    super(bearerTokenFactory, request);
+    this.cromwellWorkflowService = cromwellWorkflowService;
+    this.wsmService = wsmService;
+  }
+
+  @Override
+  public ResponseEntity<ApiWorkflowIdAndStatus> getWorkflowStatus(
+      UUID workspaceId, UUID workflowId) {
+    // Check if the user has access to the workspace.
+    wsmService.checkWorkspaceReadAccess(workspaceId, getToken().getToken());
```

Comment:
lets add unit tests for these functions.  particularly want to verify the access checking and workspace checking.